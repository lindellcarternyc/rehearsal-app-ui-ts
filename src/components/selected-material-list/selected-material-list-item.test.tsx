import * as React from 'react'
import * as renderer from 'react-test-renderer'

import SelectedMaterialListItem, { SelectedMaterialListItemProps } from './selected-material-list-item'

describe('SelectedMaterialListItem', () => {
  const material = 'test material'
  const id = 0
  const callback = jest.fn()

  it('should render with a callback', () => {
    const props: SelectedMaterialListItemProps = {
      material, id, callback
    }
    const tree = renderer.create(
      <SelectedMaterialListItem {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render without a callback', () => {
    const props: SelectedMaterialListItemProps = {
      material, id
    }
    const tree = renderer.create(
      <SelectedMaterialListItem {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})