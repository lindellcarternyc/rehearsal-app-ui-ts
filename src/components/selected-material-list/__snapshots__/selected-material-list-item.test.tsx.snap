// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedMaterialListItem should render with a callback 1`] = `
<div
  className="item"
  data-value={undefined}
  onClick={[Function]}
  role="listitem"
>
  test material
   
  <i
    aria-hidden="true"
    className="cancel icon"
    onClick={[Function]}
  />
</div>
`;

exports[`SelectedMaterialListItem should render without a callback 1`] = `
<div
  className="item"
  data-value={undefined}
  onClick={[Function]}
  role="listitem"
>
  test material
</div>
`;
